test/boost/cdc_test.cc

```diff
@@ -460,18 +465,28 @@ SEASTAR_THREAD_TEST_CASE(test_pre_image_logging) {
 
                 auto pre_image = to_bytes_filtered(*rows, cdc::operation::pre_image);
                 auto second = to_bytes_filtered(*rows, cdc::operation::update);
+                auto post_image = to_bytes_filtered(*rows, cdc::operation::post_image);
 
                 if (!enabled) {
                     BOOST_REQUIRE(pre_image.empty());
+                    BOOST_REQUIRE(post_image.empty());
                 } else {
                     sort_by_time(*rows, second);
+                    sort_by_time(*rows, pre_image);
+                    sort_by_time(*rows, post_image);
                     BOOST_REQUIRE_EQUAL(pre_image.size(), i + 1);
 
                     val = *pre_image.back()[val_index];
```

Comment:
nitpick: now that you added val2, the preimage should contain it too, you can check that...

Reply:
No, pre-image only contains columns modified. val2 is not touched by update, so not present.

Reply:
Oh, I didn't realize this until now! Wow. I'll send a separate mail about this issue.